### ISSUE TITLE
[Feature]: Support devbox.yaml

### ISSUE THREAD

**Issue by d-enk** [jetify-com/devbox#2169]
### What problem are you trying to solve?

exclude json restrictions

reopen #1590
 
see https://github.com/jetify-com/devbox/issues/1590#issuecomment-2178285560

### What solution would you like?

Support `yaml` format

### Alternatives you've considered

If you really don’t want it that much, you can at least add a path option for `devbox.yaml`
So that you can configure something like:
```shell
yq eval 'json(.)' input.yaml | devbox --path -
```

**Comment by justinTM** [comment]
@d-enk dude yes! it suddenly clicked how i've been fighting JSON quote escapes and multiline unreadability and YAML would fix this. now we just need to bully the Jetify team into adding a CLI flag to revert v0.12.0's insane breaking changes :p

**Comment by CourteousCoder** [comment]
> [@d-enk](https://github.com/d-enk) dude yes! it suddenly clicked how i've been fighting JSON quote escapes and multiline unreadability and YAML would fix this. now we just need to bully the Jetify team into adding a CLI flag to revert v0.12.0's insane breaking changes :p

Yeah this is a helpful one. I end up using an external task runner (e.g. with a `makefile`, `justfile`, `justfile`, et.c.), partially to avoid quote hell, and partially to declutter `devbox.json`.

Although, the official suggestion is to use of stand-alone script files that devbox references (e.g. an external `.scripts` directory full of different scripts for devbox to use). That also suffices for avoiding `json` quoting hell. And since devbox implements a lot of the same features as external task runners, that might be enough for you.

Another option I thought of is to extend the plugin API to hook into how a project parses devbox manifest files. That way we can just encourage community plugins as opposed to having devbox expand the scope of project maintenance.  I'll make a separate ticket for that idea and link these one-off suggestions.

**Comment by gdw2** [comment]
I wish I could put comments in my devbox.json file.

**Comment by ascknx** [comment]
> I wish I could put comments in my devbox.json file.

you already can. devbox uses json5

**Comment by AdrianOrdonezAtFanatics** [comment]
> you already can. devbox uses json5

devbox uses [tailscale/hujson](https://github.com/tailscale/hujson) which is [jwcc](https://nigeltao.github.io/blog/2021/json-with-commas-comments.html), not json5. It doesn't support things like unquoted keys or line breaks in strings.

Tested this because you got me excited that it might support json5 for multiline strings. I think that is my biggest reason for wanting `devbox.yaml` support. For now I'm using a `scripts` directory.
